timelock-api/src/main/conjure/timelock-api.yml

```diff
@@ -0,0 +1,54 @@
+types:
+  imports:
+    LeaderTime:
+      base-type: any
+      external:
+        java: com.palantir.lock.v2.LeaderTime
+    NanoTime:
+      base-type: safelong
+      external:
+        java: com.palantir.common.time.NanoTime
+    PartitionedTimestamps:
+      base-type: any
+      external:
+        java: com.palantir.lock.v2.PartitionedTimestamps
+    LockImmutableTimestampResponse:
+      base-type: any
+      external:
+        java: com.palantir.lock.v2.LockImmutableTimestampResponse
+    Lease:
+      base-type: any
+      external:
+        java: com.palantir.lock.v2.Lease
+  definitions:
+    default-package: com.palantir.atlasdb.timelock.api
+    objects:
+      ConjureStartTransactionsRequest:
+        fields:
+          requestId: uuid
+          requestorId: uuid
+          numTransactions: integer
+      ConjureStartTransactionsResponse:
+        fields:
+          immutableTimestamp: LockImmutableTimestampResponse
+          timestamps: PartitionedTimestamps
+          lease: Lease
+
+services:
+  ConjureTimelockService:
+    name: Timelock service
+    default-auth: header
+    package: com.palantir.atlasdb.timelock.api
+    base-path: /tl
+    endpoints:
+      startTransactions:
+        http: POST /st/{namespace}
+        args:
+          namespace: string
+          request: ConjureStartTransactionsRequest
+        returns: ConjureStartTransactionsResponse
+      leaderTime:
+        http: POST /lt/{namespace}
```

Comment:
Rough perspective:

1. This string does not matter in any real way. It's for routing purposes - timelock is an RPC server, it's not RESTful in any way.
2. It _will_ get sent over the wire, and at our request rates, the difference between /timelock/{namespace}/getLeaderTime and /tl/{namespace]/lt is about 1.5MB/sec.
3. Internal to the server, common prefixes are more expensive - routers typically work as a list of patterns and for each one you check which one you match - common prefixes are more expensive to match against. Ideally every endpoint we have will differ in the first character.

Reply:
So my conclusion was: for timelock endpoint, aim for short and sweet, and ideally somewhat memorable. So lt, st, ts, lo, ul, rl.

Reply:
2 is fair. Ah, okay, thanks for doing the math. 3 is also true (though I didn't think we had too many long common prefixes, other than the `startAtlasDbTransactionV(number)` ones).

I see, wasn't sure how aggressively we were going for perf. This makes sense to me.

Reply:
@j-baker please explain your reasoning here a bit more. Where does the 1.5MB/sec come from? http2 should have header compression built in, which to me means that PATHs should basically be a matter of sending some small constants

Reply:
@felixdesouza I think it already happens? `lt` will get rendered as `leaderTime` and so on as far as code is concerned.

CBOR: we could give it a shot, but uncertain if it would change things. I think it only deals with response bodies, which generally aren't large for these endpoints.

Discussed offline re: shorter paths. This is for perf. We don't know if this will necessarily be very much better, but since we're writing something new and we believe this causes little harm we'll go for it.